src/include/connection.h

```diff
@@ -209,7 +209,10 @@ struct __wt_connection_impl {
     uintmax_t optrack_pid;            /* Cache the process ID. */
 
     WT_LSN *debug_ckpt;      /* Debug mode checkpoint LSNs. */
-    uint32_t debug_ckpt_cnt; /* Checkpoint log file retention number */
+    size_t debug_ckpt_alloc; /* Checkpoint retention allocated. */
+    uint32_t debug_ckpt_cnt; /* Checkpoint retention number. */
+    /* XXX This can become a connection flag when more bits are available. */
+    bool debug_ckpt_enabled; /* Checkpoint retention enabled. */
```

Comment:
Can we just change `WT_CONNECTION_IMPL.flags` to 64 bits?

If you hate that, maybe create a new `WT_CONNECTION_IMPL.debug` field that takes this flag, plus `WT_CONN_DEBUG_CURSOR_COPY`, `WT_CONN_DEBUG_REALLOC_EXACT` and `WT_CONN_DEBUG_SLOW_CKPT`, and maybe the timing stress flags?

Reply:
Ok. I'll bite that bullet here. I prefer the `WT_CONNECTION_IMPL.debug` suggestion. I was trying to avoid too much of that work in this change but you're right that this is unpleasant.